### PR TITLE
layouts: dvorak_emacs: Add missing keys, caps word, improve LEDs.

### PR DESCRIPTION
**Pull request by Apteryks**
The following changes are made:

- Enable caps word feature
- Add the `, CAPS, SCRL, INS keys as well as L1 toggles to the base layer
- Add Mute, Bright+, Bright-, Prev, Next, Stop and Play to the AUX layer
- Wire LED 2 and 3 to caps word and scroll lock (previously unused)
- Use PERMISSIVE_HOLD strategy, to reduce errors at high typing speed
- Remove some obsolete/dead code such as KEY_TAP_FAST/KEY_TAP_SLOW

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python) (clang-formatted)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: layouts/community/ergodox/dvorak_emacs_software/keymap.c [hunk 1]
```diff
@@ -6,38 +6,28 @@
...
  * DVORAK LAYOUT (see http://djelibeibi.unex.es/dvorak/)
  * Layer 1: auxiliary keys
- * Layer 2: full qwerty layout
  *****************************************************************************************/
 
 // LAYERS
-#define BASE    0 // dvorak layout (default)
-#define AUX     1 // auxiliary keys
+#define BASE 0 // dvorak layout (default)
+#define AUX 1  // auxiliary keys
```
Comment (waffle87) [inline layouts/community/ergodox/dvorak_emacs_software/keymap.c:13]:
```suggestion
enum layers {
    BASE, // dvorak layout (default)
    AUX   // auxiliary keys
};
```

File: layouts/community/ergodox/dvorak_emacs_software/config.h
Comment (waffle87) [inline layouts/community/ergodox/dvorak_emacs_software/config.h:1]: This file is missing a license header.

File: layouts/community/ergodox/dvorak_emacs/keymap.c [hunk 2]
```diff
@@ -5,38 +5,28 @@
...
  * DVORAK LAYOUT (see http://djelibeibi.unex.es/dvorak/)
  * Layer 1: auxiliary keys
- * Layer 2: full qwerty layout
  *****************************************************************************************/
 
 // LAYERS
-#define BASE    0 // dvorak layout (default)
-#define AUX     1 // auxiliary keys
+#define BASE 0 // dvorak layout (default)
+#define AUX 1  // auxiliary keys
```
Comment (waffle87) [inline layouts/community/ergodox/dvorak_emacs/keymap.c:12]:
```suggestion
enum layers {
    BASE, // dvorak layout (default)
    AUX   // auxiliary keys
};
```

File: layouts/community/ergodox/dvorak_emacs/config.h
Comment (waffle87) [inline layouts/community/ergodox/dvorak_emacs/config.h:1]: This file is missing a license header.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: layouts/community/ergodox/dvorak_emacs_software/config.h
Comment (Apteryks) [reply layouts/community/ergodox/dvorak_emacs_software/config.h:1]: Added.

File: layouts/community/ergodox/dvorak_emacs/config.h
Comment (Apteryks) [reply layouts/community/ergodox/dvorak_emacs/config.h:1]: Add SPDX + copyright.

File: layouts/community/ergodox/dvorak_emacs/keymap.c (hunk 2 above)
Comment (Apteryks) [reply layouts/community/ergodox/dvorak_emacs/keymap.c:12]: Done! Thanks for the review.

**Comment by Apteryks** [comment]
@waffle87 I've reopened a new PR here with your suggested changes implemented: https://github.com/qmk/qmk_firmware/pull/23741

**Pull request closed without merging**